server/api.js

```diff
@@ -1,13 +1,45 @@
-const express = require('express');
+const express = require("express");
 const router = express.Router();
-
+const bodyParser = require("body-parser");
+const _ = require("lodash");
 const auth = require("./auth");
 
+const { Department } = require("./db/models/department");
```

Comment:
Since we are registering models in mongoose, these lines can be replaced by:

```
const mongoose = require("mongoose");
const Department = mongoose.model("Department");
const Professor = mongoose.model("Professor");
```

Reply:
That is what I tried at first. But it gives some kind of "model not found" errors. I guess it happens because the models haven't been registered in mongoose, but they are trying to be accessed here. This was just a dirty fix for that. I will try to see if requiring `models/index.js` at the start helps or not.

Reply:
A quick look at ‘models/index.js’ shows that all models except the Student model will be registered in mongoose when the server starts. There is no need to require ‘models/index.js’ here has it has already been required in ‘api.js’

Reply:
`models/index.js` hasn't been required in `api.js`. But I just did that locally, and your suggestion is working. I'll push the updated code now.